core/DataTable.php

```diff
@@ -1291,6 +1299,15 @@ public function getSerialized($maximumRowsInDataTable = null,
             $subtableId = 0;
             throw new Exception("Maximum recursion level of " . self::$maximumDepthLevelAllowed . " reached. Maybe you have set a DataTable\Row with an associated DataTable belonging already to one of its parent tables?");
         }
+
+        // gather metadata before filters are called, so their metadata is not stored in serialized form
+        $metadata = $this->getAllTableMetadata();
+        foreach ($metadata as $key => $value) {
+            if (!is_scalar($value) && !is_string($value)) {
```

Comment:
@diosmosis quickly looked through some archivers and dataArrays and it seems like there's usually no metadata on the dataTable so that should be fine. Haven't checked search engine plugin or so. I wondered if we really want to persist by default automatically all metadata or whether we should maybe require to specify whether some/which metadata should be persisted or not? Not sure if it would be always wanted to persist all information. I guess we could also just leave it like this, and eventually change it later when need comes up to whitelist specific metadata.

Reply:
I could add a 'metadata_to_serialize' metadata, though maybe that's just too much meta...

Reply:
I think it's not needed for now. Initially thought it was a good idea, but then realising it's probably over-engineering in the  sense that we seem to not have a use case for it just yet and we can always add it later. I reckon be good to keep it simple for now. 

Couldn't find any table that sets metadata during the archiving in core and our plugins. Don't think there would be any third party plugins out there that do it, and if so, it likely doesn't even cause an issue. I suppose if you set metadata, maybe it's what people expect that it gets persisted 👍